lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,37 @@
+module Semian
+  class AtomicEnum < AtomicInteger #:nodoc:
+    undef :increment_by
+    undef :increment
+
+    module AtomicEnumSharedImplementation
+      def initialize(name, permissions, symbol_list)
+        super(name, permissions)
+        initialize_lookup(symbol_list)
+      end
+
+      def value
+        @num_to_sym.fetch(super)
+      end
+
+      def value=(sym)
+        super(@sym_to_num.fetch(sym))
```

Comment:
:+1: on `#fetch` :-)

It would make sense to `alias_method` this to `#transition` or something similar, which would make the Circuit Breaker code easier to read. Yes / no?